src/pages/ReservationDetails.tsx

```diff
@@ -13,146 +29,301 @@ export default function ReservationApproval() {
+      });
+  }, []);
+
+  const isIn12Hours = (timestamp: number): boolean => {
+    timestamp = timestamp * 1000;
     const now = new Date();
-    const difference = (timestamp - now.getTime()) * 60 * 60 * 1000;
-    if (difference > 0 && difference <= 12) {
-      return true;
-    }
-    return false;
-  }
+    const difference = (timestamp - now.getTime()) / 60 / 60 / 1000;
+    const a = difference > 0 && difference <= 12;
+    return a;
+  };
 
-  return (
-    <div className='booking-summary-container'>
-      <h3>Reservation summary</h3>
-      <div className='booking-summary-date_person-container'>
-        <div>
-          <p className='booking-summary-label'>Date</p>
-          <DatePicker
-            value={reservation.time?.from ?? 0}
-            onChange={() => {
-              return;
-            }}
-            renderInput={params => (
-              <TextField {...params} size='small' fullWidth disabled />
-            )}
-          />
-        </div>
+  const cancelReservation = async () => {
+    setCancelButtonLoading(true);
 
-        <div>
-          <p className='booking-summary-label'># of persons</p>
-          <TextField
-            id='outlined-number'
-            type='number'
-            value={2}
-            InputProps={{
-              readOnly: true,
-            }}
-            size='small'
-            disabled
-          />
-        </div>
-      </div>
+    window.confirm('Are you sure you want to cancel your reservation?') &&
+      (await reservationApi
+        .deleteReservation(reservationId ?? '')
+        .then(() => {
+          successAlert('Reservation canceled!');
+          setReservation(undefined);
+          setShowReservation(false);
+        })
+        .catch(() => {
+          errorAlert('Something went wrong canceling your reservation');
+        }));
+    setCancelButtonLoading(false);
+  };
 
-      <p className='booking-summary-label'>Time</p>
-      <div className='booking-summary-time-picker-container'>
-        <TimePicker
-          onChange={() => {
-            return;
-          }}
-          readOnly
-          value={reservation.time?.from ?? 0}
-          renderInput={params => (
-            <TextField {...params} size='small' disabled />
-          )}
-          ampm={false}
-        />
-        <TimePicker
-          value={reservation.time?.to ?? 0}
-          readOnly
-          onChange={() => {
-            return;
-          }}
-          renderInput={params => (
-            <TextField {...params} size='small' disabled />
-          )}
-          ampm={false}
-        />
-      </div>
+  const confirmReservation = async () => {
+    setConfirmButtonLoading(true);
+    await reservationApi
+      .putReservation(
+        { confirmed: true },
+        reservation?.id ?? '',
+        confirmationToken ?? '',
+      )
+      .then(() => {
+        successAlert('Reservation confirmed!');
+        reservation !== undefined ? (reservation.confirmed = true) : null;
```

Comment:
```suggestion
        setReservation(reservation);
```
Spricht da was dagegen? Ansonsten musst du es so machen:
```tsx
setReservation({...reservation, confirmed: true})
```
weil man darf einen State nich direkt verändern.

Reply:
Als deine Suggestion funktioniert nicht, weil dann des confirmed, noch nicht auf true ist.
Man kann entweder nochmal die Reservation vom Server holen oder man kann einfach auf dem Client die reservation auf true setzen, dass musst du entscheiden